parsons/google/google_cloud_storage.py

```diff
@@ -327,17 +323,18 @@ def upload_table(
             # CSVView. Once any transformations are made, the Table.table
             # becomes a different petl class
             if isinstance(table.table, petl.io.csv_py3.CSVView):
```

Comment:
This change looks okay to me, I'm curious about the reasoning? That will help me understand how big the breaking change would be - right now this code is so similar that it's hard to tell how it might cause problems for downstream users.

Reply:
The reasoning for the change is that something about how our Parsons operator is working is changing the attributes of the CSVView, and we just need to do the `to_csv` instead of getting the filename. It would presumably break for someone if they want it to fail when encountering an Attribute Error. I assume this is unlikely and that this probably wouldn't really break anything for anyone, but technically if someone is depending on getting an AttributeError in some case, this would mean they instead get an actual table. Not sure why that would be a problem, but trying to be proactive!

Reply:
Okay gotcha, that makes sense. Yeah, it's hard to imagine someone depending on getting an error there - there's no reason to call the function at all if you error out there. It hasn't uploaded the data to the blob yet, and it's not saving anything on the object that you can access, so there's no reason to call the function and catch the attribute error unless you're checking if `table.table.source.filename` exists, doing something, then calling the function again. But that would be a very weird pattern, like you could literally just check if `table.table.source.filename` exists directly. So yes, this is technically a breaking change but I think the chances of anyone being bothered by it are very very small so we are okay to merge it into main instead of major release.